### ISSUE TITLE
Making the docs for blueprints more explicit about usecases with some examples.

### ISSUE THREAD

**Issue by berkio3x** [pallets/flask#3675]
A lot of people have confusion about the use case of blueprints in Flask. This is also based on the first-hand experience of introducing flask to a lot of developers. some random examples when you search on :
- [example 1](https://www.reddit.com/r/flask/comments/bp9j23/eli5_the_blueprint_concept_of_flask/)
- [example 2](https://www.reddit.com/r/flask/comments/bamhop/can_someone_better_explain_flask_blueprints_to_me/)
- [example 3](https://www.reddit.com/r/flask/comments/6nf9m1/af_blueprints_eli5/) 
- [example 4](https://www.reddit.com/r/flask/comments/d9rghe/i_dont_entirely_understand_the_purpose_of/)

**NOTE**: _I am happy and willing to do these changes/add examples myself, just wanted to know the views of others on this._

Afaik these confusions are from people who come straight to flask for web app development without having any prior experience. Even for such cases, we should try to be clear about what can/cannot be done by blueprints (including examples where possible)

1. The first point in the official doc tries to make the point of modularity but the wording might be a bit confusing.

>Factor an application into a set of blueprints. This is ideal for larger applications; a project could instantiate an application object, initialize several extensions, and register a collection of blueprints.

If we change this to (or anything else, the idea is just to make it clear):

_Factor an application into a set of related logical components to modularize the code. This is ideal for larger applications; a project could instantiate an application object and instead of stuffing every route/view on the main app object, which makes code hard to maintain/extend,  create and register a collection of blueprints on `app`._

2. Add an example for this:
>Register a blueprint on an application at a URL prefix and/or subdomain. Parameters in the URL prefix/subdomain become common view arguments (with defaults) across all view functions in the blueprint.

3. Make templating behaviour of blueprints(templating is where blueprint really shines) explicit with example. This is mentioned in the official doc but lacks an example so it is not easily understood by many.

>Register a blueprint multiple times on an application with different URL rules.

we may add some example like the one below to make it clear.

An application that is used for the online digital library should also provide access to books, authors, billings. As soon as the library blueprint is registered on the app, the app has access to all the features of the library under  different namespaces but with similar behaviours.
```

from flask import Blueprint

library = Blueprint("library", __name__)

@library.route("/books")
def book():
    return books()

@library.route("/authors")
def author():
    return authors()

@library.route("/payment")
def payment(year=None):
    # some payment  logic .....

app.register_blueprint(library, url_prefix="/space")
app.register_blueprint(library, url_prefix="/animal-kingdoms")
app.register_blueprint(library, url_prefix="/geography")

```

**Comment by davidism** [comment]
@rochacbruno had some ideas on enhancing the docs about the app factory pattern and blueprints too. Happy to consider PRs!

**Comment by berkio3x** [comment]
Ok. I am taking this on me.

**Comment by berkio3x** [comment]
@rochacbruno I am curious about what enhancements you wanted in Blueprints docs?

**Comment by davidism** [comment]
@berkio3x were you still planning to submit a PR with your suggested changes?

**Comment by berkio3x** [comment]
@davidism yes, I have a draft prepared , need to clean it up, caught up in work. will submit the pr this weekend :)
